pkg/apis/core/validation/validation.go

```diff
@@ -6000,6 +6003,81 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
 	return allErrs
 }
 
+func validateWindowsHostProcessPod(podSpec *core.PodSpec, fieldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	containerCount := 0
+	hostProcessContainerCount := 0
+
+	var podHostProcess *bool
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.WindowsOptions != nil && podSpec.SecurityContext.WindowsOptions.HostProcess != nil {
+		podHostProcess = podSpec.SecurityContext.WindowsOptions.HostProcess
+	}
+
+	hostNetwork := false
+	if podSpec.SecurityContext != nil {
+		hostNetwork = podSpec.SecurityContext.HostNetwork
+	}
+
+	podshelper.VisitContainersWithPath(podSpec, fieldPath, func(c *core.Container, cFieldPath *field.Path) bool {
+		containerCount++
```

Comment:
separating the checks for "do container-level and pod-level hostProcess agree?" and "is this a hostProcess container?" would eliminate a lot of branching and make this shallower/easier-to-follow:

```go
if podHostProcess != nil && containerHostProcess != nil && *podHostProcess != *containerHostProcess {
	errMsg := fmt.Sprintf("pod and container windowsOptions.hostProcess must match if specified, pod was %v, container was %v", *podHostProcess, *containerHostProcess)
	allErrs = append(allErrs, field.Forbidden(cFieldPath.Child("securityContext","windowsOptions","hostProcess"), errMsg))
}

switch {
case containerHostProcess != nil && *containerHostProcess:
  // explicit container-level hostProcess=true
  hostProcessContainerCount++
case containerHostProcess == nil && podHostProcess != nil && *podHostProcess:
  // no container-level setting, explicit pod-level hostProcess=true
  hostProcessContainerCount++
}
```